packages/cli/src/util/oauth.ts

```diff
@@ -380,22 +398,59 @@ function canParseURL(url: string) {
   }
 }
 
-interface VercelAccessToken extends JWTPayload {
-  team_id?: string;
+/** @see https://datatracker.ietf.org/doc/html/rfc7662#section-2.2 */
+interface Token {
+  /**
+   * Integer timestamp, measured in the number of seconds
+   * since January 1 1970 UTC, indicating when this token will expire.
+   */
+  exp: number;
+  /** Whether or not the presented token is active. */
+  active: boolean;
+}
+
+interface AccessToken extends Token {
+  /** The authorizing principal's team. */
+  team_id: string;
+  token_type: 'access_token';
+}
+interface RefreshToken extends Token {
+  token_type: 'refresh_token';
 }
 
-export async function verifyJWT(
-  token: string
-): Promise<[Error] | [null, VercelAccessToken]> {
+/**
+ * Inspects and returns the content of a given token.
+ * If the token is invalid, an {@link InspectionError} is returned.
+ */
+export async function inspectToken<
+  T extends TokenSet['access_token'] | NonNullable<TokenSet['refresh_token']>,
+  Payload extends T extends TokenSet['refresh_token']
+    ? RefreshToken
+    : AccessToken,
+>(token: T): Promise<[InspectionError] | [null, Payload]> {
   try {
+    // TODO: Use an [Introspection Endpoint](https://datatracker.ietf.org/doc/html/rfc7662)
+
     const JWKS = createRemoteJWKSet((await as()).jwks_uri);
-    const { payload } = await jwtVerify<VercelAccessToken>(token, JWKS, {
+    const { payload } = await jwtVerify<Payload>(token, JWKS, {
       issuer: 'https://vercel.com',
       audience: ['https://api.vercel.com', 'https://vercel.com/api'],
     });
+
+    // TODO: Remove override when we have an introspection endpoint
+    payload.active ??= Boolean(
+      payload.exp && payload.exp < Math.floor(Date.now() / 1000)
+    );
+
+    if (!payload.active) {
+      throw new InspectionError('Token is not active.');
+    }
```

Comment:
Is this necessary, or could it be removed?

IIUC, the only hard dependency we currently have is `team_id`?

Reply:
This _is_ necessary, this is how we know to perform the token refresh. I updated the format to match a future implementation of an [Introspection Endpoint](https://datatracker.ietf.org/doc/html/rfc7662#section-2.2)